File: javascript/rna-transcription/rna-transcription.js
//
// This is only a SKELETON file for the 'RNA Transcription' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const toRna = (dna) => {
  const nucleotides = {G: "C" , C: "G" , T: "A", A: "U"}
  return dna.split("").map(x => nucleotides[x]).join("")
};

File: clojure/rna-transcription/src/rna_transcription.clj
(ns rna-transcription)

(require '[clojure.string :as string])

(defn process [dna]
  (map {\C "G" \G "C" \T "A" \A "U"} dna))

(defn to-rna [dna]
  (let [p (->  (process dna) string/join)]
    (assert (= (count dna) (count p)))
    p))
